src/status_im2/contexts/chat/lightbox/zoomable_image/view.cljs

```diff
@@ -204,76 +203,61 @@
-           props                                   {:pan-x-enabled? (reagent/atom false)
-                                                    :pan-y-enabled? (reagent/atom false)
-                                                    :focal-x        (reagent/atom nil)
-                                                    :focal-y        (reagent/atom nil)}
-           rescale                                 (fn [value exit?]
-                                                     (utils/rescale-image value
-                                                                          exit?
-                                                                          dimensions
-                                                                          animations
-                                                                          props))]
-       (rn/use-effect (fn []
-                        (js/setTimeout #(reset! set-full-height? true) 500)))
-       (when platform/ios?
-         (utils/handle-orientation-change curr-orientation focused? dimensions animations props)
-         (utils/handle-exit-lightbox-signal exit-lightbox-signal
-                                            index
-                                            (anim/get-val (:scale animations))
-                                            rescale
-                                            set-full-height?))
-       (utils/handle-zoom-out-signal zoom-out-signal index (anim/get-val (:scale animations)) rescale)
-       [:f>
-        (fn []
-          (let [tap (tap-gesture #(on-tap portrait?))
-                double-tap
-                (double-tap-gesture dimensions animations rescale transparent? #(on-tap portrait?))
-                pinch
-                (pinch-gesture dimensions animations props rescale transparent? #(on-tap portrait?))
-                pan-x (pan-x-gesture dimensions animations props rescale)
-                pan-y (pan-y-gesture dimensions animations props rescale)
-                composed-gestures (gesture/exclusive
-                                   (gesture/simultaneous pinch pan-x pan-y)
-                                   (gesture/exclusive double-tap tap))]
-            [gesture/gesture-detector {:gesture composed-gestures}
-             [reanimated/view
-              {:style (style/container dimensions
-                                       animations
-                                       @set-full-height?
-                                       (= curr-orientation orientation/portrait))}
-              [reanimated/fast-image
-               {:source    {:uri (:image content)}
-                :native-ID (when focused? :shared-element)
-                :style     (style/image dimensions animations (:border-value args))}]]]))]))])
+  [{:keys [image-width image-height content message-id]} index render-data]
+  (let [state                      (utils/init-state)
```

Comment:
The call to `utils/init-state` should happen outside the renderer.

Reply:
@ilmotta for some reason moving `utils/init-state` outside of the render causes the component to not behave properly. Any thoughts on this

```
(let [state                      (utils/init-state)]
    (fn []
      (let [
            shared-element-id          (rf/sub [:shared-element-id])
...
```

Reply:
I've noticed in other PRs that this problem appeared a couple times, i.e. if the state is not initialized inside the renderer something doesn't work. The root cause can be quite complicated to find due to the amount of stateful things going on.

I know it's not the best answer, but my first approach would be to rewrite a little bit this `zoomable-image` component and maybe others. For instance, the calls to `utils/handle-orientation-change` or `handle-exit-lightbox-signal`, `handle-zoom-out-signal` cause mutations, and they are called on every re-render of the component. We know we should almost never do this.

Hence, the problems you saw may be caused by multiple issues in the code, and they all affect each other. I would need to get my hands dirty to find out why, sorry I don't have anything more concrete than this.

Reply:
Oh I see. I have tried to change a few things, but did not work. It could be even more complex than this. The problem could be coming from the lightbox component. I will investigate further in another issue: https://github.com/status-im/status-mobile/issues/16068

Reply:
Oh I see. I have tried to change a few things, but did not work. It could be even more complex than this. The problem could be coming from the lightbox component. I will investigate further in another issue: https://github.com/status-im/status-mobile/issues/16068